tests/query_runner/test_google_spreadsheets.py

```diff
@@ -104,6 +109,40 @@ def test_parse_query(self):
         parsed = parse_query("key|0")
         self.assertEqual(("key", 0), parsed)
 
+    def test_parse_query_ignored(self):
+        parsed = parse_query("key")
+        self.assertEqual(("key", 0), parsed)
+
+        parsed = parse_query("key|")
+        self.assertEqual(("key", 0), parsed)
+
+        parsed = parse_query("key|1|")
+        self.assertEqual(("key", 0), parsed)
```

Comment:
Confusingly, it seems that the sheet numbers are 1-indexed, but the returned values are 0-indexed. We might want to add a comment

Reply:
@guidopetri  Here, when calling the parse_query function with the string as "key|1|" then following execution happens inside the function.
Inside the file `redash.query_runner.google_spreadsheets.parse_query`
- At L104 the value of the variable `values` will be `["key", "1", ""]`
- At L107 there is a specific check if the length of `values` is 2 then only process the query further else return the default values which in this will be the value of the variable `key`(At L105) and `worksheet_num_or_title`(At L106)
- For passed query(`"key|1|"`) the total length of `values` will be 3 and so the default values (`tuple("key", 0)`) is returned.

With this cleared, the check on L120 in the test file passes, so I believe the sheets are 0-indexed.